File: insertionsort.py
import random
def insertionSort(array):
    for i in range (1, len(array)):
        j = i - 1
        while j >= 0 and array[j + 1] < array[j]:
            temp = array[j + 1]
            array[j + 1] = array[j]
            array[j] = temp
            j = j - 1
        
    return array


def insert_sort(nums):
    for tb in range(1, len(nums)):
        move = nums[tb]
        hand = tb - 1
        while hand >= 0 and nums[hand] > move:
            nums[hand+1] = nums[hand]
            hand -= 1
        nums[hand+1] = move
    return nums


arr = [5, 4, 3, 2, 1]
arr1 = list(range(20))
random.shuffle(arr1)

print(arr1)
insert_sort(arr1)
print(arr1)

File: insertionsort.c
#include <stdio.h>
#include <stdlib.h>

void insertionSort(int array[], int length) {
    for (int i = 1; i < length; i++) {
        int j = i - 1;
        int key = array[i];
        while (j >= 0 && array[j] > key) {
            array[j + 1] = array[j];
            j = j - 1;
        }
        array[j + 1] = key;
    }
}

int main() {
    int arr1[20];
    for (int i = 0; i < 20; i++) {
        arr1[i] = i + 1;
    }
    // Shuffle the array
    for (int i = 19; i > 0; i--) {
        int j = rand() % (i + 1);
        int temp = arr1[i];
        arr1[i] = arr1[j];
        arr1[j] = temp;
    }

    // Print the shuffled array
    printf("Shuffled array:\n");
    for (int i = 0; i < 20; i++) {
        printf("%d ", arr1[i]);
    }
    printf("\n");

    // Sort the array using insertion sort
    insertionSort(arr1, 20);

    // Print the sorted array
    printf("Sorted array:\n");
    for (int i = 0; i < 20; i++) {
        printf("%d ", arr1[i]);
    }
    printf("\n");

    return 0;
}

